### ISSUE TITLE
Flake on RasterThreadMerger.RunExpiredTasksWhileFirstTaskUnMergesThreads

### ISSUE THREAD

**Issue by iskakaushik** [flutter/flutter#66922]
See: https://ci.chromium.org/p/flutter/builders/prod/Linux%20Fuchsia/5598

cc: @lucasradaelli, @dnfield

**Comment by dnfield** [comment]
This failure doesn't make sense for the linked commit. The linked commit was a change to a11y. The failure is in the RasterThreadMerger tests.

```
[ RUN      ] RasterThreadMerger.RunExpiredTasksWhileFirstTaskUnMergesThreads
../../flutter/fml/raster_thread_merger_unittests.cc:566: Failure
Expected equality of these values:
  fml::MessageLoop::GetCurrentTaskQueueId()
    Which is: 168
  qid_platform
    Which is: 167
[  FAILED  ] RasterThreadMerger.RunExpiredTasksWhileFirstTaskUnMergesThreads (0 ms)
```

**Comment by iskakaushik** [comment]
The test failures seem like a flake. De-escalating to P3.

**Comment by zanderso** [comment]
This happened again here https://ci.chromium.org/ui/p/flutter/builders/prod/Linux%20Fuchsia%20FEMU/3114/overview

**Comment by iskakaushik** [comment]
happened again on: https://ci.chromium.org/ui/p/flutter/builders/try/Linux%20Fuchsia%20FEMU/2780/overview

**Comment by dnfield** [comment]
Seems like it's only happening on fuschia.  /Cc @arbreng @chaselatta

**Comment by iskakaushik** [comment]
Seen again: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket.appspot.com/8845516146557485760/+/u/FEMU_Test/Run_FEMU_Test_fml_tests/stdout

**Comment by arbreng** [comment]
https://github.com/flutter/engine/pull/26542 fixed this, but it should be reverted after https://github.com/flutter/engine/pull/26193 lands

**Comment by zanderso** [comment]
It looks like this is still happening: 

https://ci.chromium.org/ui/p/flutter/builders/prod/Linux%20Fuchsia%20FEMU/4109/overview
https://ci.chromium.org/ui/p/flutter/builders/prod/Linux%20Fuchsia%20FEMU/4123/overview
